src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs

```diff
@@ -60,38 +60,75 @@ protected override void ExecuteCore()
-                    {
-                        var reference = new TaskItem(dll);
-
-                        reference.SetMetadata(MetadataKeys.ExternallyResolved, "true");
-                        reference.SetMetadata(MetadataKeys.Private, "false");
-
-                        //  TODO: Once we work out what metadata we should use here to display these references grouped under the targeting pack
-                        //  in solution explorer, set that metadata here.These metadata values are based on what PCLs were using.
-                        //  https://github.com/dotnet/sdk/issues/2802
-                        reference.SetMetadata("WinMDFile", "false");
-                        reference.SetMetadata("ReferenceGroupingDisplayName", targetingPack.ItemSpec);
-                        reference.SetMetadata("ReferenceGrouping", targetingPack.ItemSpec);
-                        reference.SetMetadata("ResolvedFrom", "TargetingPack");
-                        reference.SetMetadata("IsSystemReference", "true");
-
-                        referencesToAdd.Add(reference);
-                    }
-
-                    var platformManifestPath = Path.Combine(targetingPackRoot, "build", "netcoreapp3.0",
-                        targetingPack.GetMetadata(MetadataKeys.PackageName) + ".PlatformManifest.txt");
+                    string targetingPackFormat = targetingPack.GetMetadata("TargetingPackFormat");
+                    string targetingPackTargetFramework = targetingPack.GetMetadata("TargetFramework");
 
-                    if (File.Exists(platformManifestPath))
+                    if (targetingPackFormat.Equals("NETStandardLegacy", StringComparison.OrdinalIgnoreCase))
                     {
-                        platformManifests.Add(new TaskItem(platformManifestPath));
+                        string targetingPackAssetPath = Path.Combine(targetingPackRoot, "build", targetingPackTargetFramework, "ref");
+
+                        foreach (var dll in Directory.GetFiles(targetingPackAssetPath, "*.dll"))
+                        {
+                            var reference = new TaskItem(dll);
+                            reference.SetMetadata(MetadataKeys.Private, "false");
+                            reference.SetMetadata("Visible", "false");
+                            reference.SetMetadata(MetadataKeys.NuGetPackageId, targetingPack.GetMetadata(MetadataKeys.PackageName));
+                            reference.SetMetadata(MetadataKeys.NuGetPackageVersion, targetingPack.GetMetadata(MetadataKeys.PackageVersion));
+
+                            if (!Path.GetFileName(dll).Equals("netstandard.dll", StringComparison.OrdinalIgnoreCase))
+                            {
+                                reference.SetMetadata("Facade", "true");
+                            }
+
+                            referencesToAdd.Add(reference);
```

Comment:
Extrasct some methods here. These large blocks in if and else are hard to follow.